### PR TITLE
Implement Base.merge

### LINKED ISSUE #40: More alignment operations
A couple more alignment operations are sometimes useful:
* `prepend(x, y)`: return knots from `x` until `y` becomes active; from then onward only return knots from `y`.
* `Base.merge(x1,  x2, ...)`: return knots from all nodes. Where there are knots from multiple nodes simultaneously, return the value which occurs in the last collection.

**NB ** Should decide whether we want to allow automatic type promotion here. Alternatively we may need a `cast` node to turn a node into one with a larger value type. [Such a node shouldn't need to copy data at evaluation time]

### PR DESCRIPTION
**Pull request by tpgillam**
Closes #40

### CODE REVIEW

File: test/ops/align.jl [hunk 1]
```diff
@@ -166,3 +167,53 @@ end
...
+    @test merge(n1, n1, n1, n1) === n1
+
+    # More generally, if a node appears multiple times, we only need to keep the _last_
+    # occurrence.
+    @test merge(n2, n1, n2) === merge(n1, n2)
+    @test merge(n1, n2, n1, n2) === merge(n1, n2)
+    @test merge(n1, n2, n3, n2, n1, n1, n2, n1, n2) === merge(n3, n1, n2)
+
+    # Merging constants should give a constant.
+    # NB: the first argument has to be a node, so as to
```
Comment (tpgillam) [inline test/ops/align.jl:185]: FIXME

File: src/ops/align.jl [hunk 2]
```diff
@@ -310,3 +310,101 @@ function count_knots(x)
...
+    # consideration for each series.
+    il = 1
+    ir = 1
+
+    # Index into the output.
+    j = 1
+
+    # Loop until we exhaust inputs.
+    # @inbounds while (il <= nl || ir <= nr)
+    while (il <= nl || ir <= nr)
```
Comment (tpgillam) [inline src/ops/align.jl:348]: FIXME

File: src/ops/align.jl [hunk 3]
```diff
@@ -310,3 +310,101 @@ function count_knots(x)
...
+    # Store indices into the inputs. The index represents the next time point for
+    # consideration for each series.
+    il = 1
+    ir = 1
+
+    # Index into the output.
+    j = 1
+
+    # Loop until we exhaust inputs.
+    # @inbounds while (il <= nl || ir <= nr)
```
Comment (willtebbutt) [inline src/ops/align.jl:347]:
```suggestion
```

File: src/ops/align.jl [hunk 4]
```diff
@@ -310,3 +310,101 @@ function count_knots(x)
...
+    input_l::Block,
+    input_r::Block,
+) where {T}
+    # This is a very simplified version of Union alignment, since we do not need to keep
+    # track of any previous values.
+
+    # Fast-paths — any emptiness of inputs means that no merging is required.
+    isempty(input_l) && return input_r
+    isempty(input_r) && return input_l
+
```
Comment (tpgillam) [inline src/ops/align.jl:330]: If identical times return `input_r`

**Review by willtebbutt — APPROVED** [review]
LGTM

**Pull request merged**